src/components/MailingList/MailingList.jsx

```diff
@@ -30,51 +34,106 @@ const EmailSubmitButton = styled.input`
 
-export default function MailingList() {
-  const [address, setAddress] = useState("");
-  const [result, setResult] = useState(null);
+const LoadingAnimation = styled.span`
+    ${(props) => (props.active ? "animation: spin 2s linear infinite;" : "")}
+    @keyframes spin {
+        0% {
+            transform: rotate(0deg);
+        }
+        100% {
+            transform: rotate(360deg);
+        }
+    }
+    display: flex;
+    justify-content: center;
+    align-items: center;
+`;
 
-  async function addEmailToList(e) {
-    e.preventDefault();
-    setResult(null);
+const READY_STATUS = "ready";
+const LOADING_STATUS = "loading";
+const SUCCESS_STATUS = "success";
+const FAILURE_STATUS = "failure";
 
-    // make email post request
-    const response = await fetch('https://api.michhackers.com/email/add', {
-      method: 'POST',
-      headers: {
-        "Content-Type": "application/json"
-      },
-      body: JSON.stringify({
-        email: address
-      })
-    });
+const statusToTitle = {};
+statusToTitle[READY_STATUS] = "Send";
+statusToTitle[LOADING_STATUS] = "Sending...";
+statusToTitle[SUCCESS_STATUS] = "Sent!";
+statusToTitle[FAILURE_STATUS] = "Failed to send";
 
-    // handle error
-    if (response.status !== 200) {
-      setResult(`An error occured: ${response.status} ${response.statusText}`);
-      return;
-    }
+export default function MailingList() {
+    const [address, setAddress] = useState("");
+    const [status, setStatus] = useState(READY_STATUS);
+    const [disabled, setDisabled] = useState(false);
 
-    // handle successful response
-    setAddress("");
-    setResult("Successfully added!");
-  }
+    async function addEmailToList(e) {
+        e.preventDefault();
+        setStatus(LOADING_STATUS);
+        setDisabled(true);
+        //make email post request
+        const response = await fetch("https://api.michhackers.com/email/add", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json"
+            },
+            body: JSON.stringify({
+                email: address,
+            }),
+        });
+
+        // handle error
+        if (response.status !== 200) {
+            setStatus(FAILURE_STATUS);
+            setDisabled(false);
+            console.error(
+                `An error occured: ${response.status} ${response.statusText}`
+            );
+            return;
+        }
+        setStatus(SUCCESS_STATUS);
+        setAddress("");
+        setDisabled(false);
+    }
 
-  return (
-    <EmailForm onSubmit={addEmailToList}>
-      {result && <StaticP>{result}</StaticP>}
-      <EmailInputBox
-        type="email"
-        value={address}
-        onChange={e => setAddress(e.target.value)}
-        placeholder="michiganhackers@umich.edu"
-        required
-      />
-      <EmailSubmitButton type="submit" value="Join" />
-    </EmailForm>
-  );
+    return (
+        <EmailForm onSubmit={addEmailToList}>
+            <EmailInputBox
```

Comment:
It could use a Toast message to indicate completeness or to spit out the error message from the server.